### PR TITLE
#1928 fix color bom

### PR DESCRIPTION
**Pull request by ArmaanMehra**
## Changes

- In the BOM Table, material status colors have been updated and made less harsh (Ordered - yellow, Unordered - maroon, Shipped - blue, Received - green).
- The assembly row color is now the same dark grey we use for pageblock backgrounds.
- Status buttons are made to be the same size, accounting for differences in window size as well
- Reformatted and abstracted BOMTableCustomCells.tsx to have a method that gets us the color of the button based on material status, and another to get the chip style based on material status.

## Notes

I believe the status colors are fine, but not sure if assembly line color is fine for light mode.

## Test Cases

No test cases needed (checked with tech lead Aaryan).

## Screenshots
small window
large window

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # 1928

### CODE REVIEW

**Review by Aaryan1203 — APPROVED** [review]
This looks great! And I think the colors look nice as well. @RChandler234 what do you think?

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx [hunk 1]
```diff
@@ -16,13 +16,13 @@ export const renderStatusBOM = (params: GridRenderCellParams) => {
   const status = params.value;
   const color =
     status === MaterialStatus.Ordered
-      ? 'orange'
+      ? '#dba63e'
       : status === MaterialStatus.Unordered
-      ? 'red'
+      ? '#ba782a'
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx:21]: can you make unordered, more maroon

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Could you also make all the status chips the same size while you're in here

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx [hunk 2]
```diff
@@ -16,16 +16,24 @@ export const renderStatusBOM = (params: GridRenderCellParams) => {
...
+      ? '#a63737'
       : status === MaterialStatus.Received
-      ? 'green'
+      ? '#2a712a'
       : status === MaterialStatus.Shipped
-      ? 'yellow'
+      ? '#1b537a'
       : 'grey';
+
+  const boxStyle = {
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx:28]: can you rename this to bomStatusChipStyle and move it above this method

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx [hunk 3]
```diff
@@ -16,16 +16,24 @@ export const renderStatusBOM = (params: GridRenderCellParams) => {
   const status = params.value;
   const color =
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/BOMTableCustomCells.tsx:17]: can you also make a helper method that determines color from MaterialStatus

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by ArmaanMehra** [comment]
@RChandler234 @Aaryan1203 all changes requested have been resolved! Please take a look when y'all can

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**